src/scf/mixing.jl

```diff
@@ -173,65 +174,39 @@ real space using a GMRES. Either the full kernel (`RPA=false`) or only the Hartr
     verbose::Bool = false  # Run the GMRES verbosely
 end
 
-@timing "χ0Mixing" function mix(mixing::χ0Mixing, basis, δF_tot::RealFourierArray,
-                                δF_spin=nothing; ρin, ρ_spin_in, kwargs...)
-    T = eltype(δF_tot)
+@views @timing "χ0Mixing" function mix(mixing::χ0Mixing, basis, δF; ρin, kwargs...)
+    T = eltype(δF)
     n_spin = basis.model.n_spin_components
     @assert basis.model.spin_polarization in (:none, :spinless, :collinear)
 
     # Initialise χ0terms and remove nothings (terms that don't yield a contribution)
-    χ0applies = [χ0(basis; ρin=ρin, ρ_spin_in=ρ_spin_in, kwargs...) for χ0 in mixing.χ0terms]
+    χ0applies = [χ0(basis; ρin=ρin, kwargs...) for χ0 in mixing.χ0terms]
     χ0applies = [apply for apply in χ0applies if !isnothing(apply)]
 
     # If no applies left, do not bother running GMRES and directly do simple mixing
-    isempty(χ0applies) && return mix(SimpleMixing(α=mixing.α), basis, δF_tot, δF_spin)
+    isempty(χ0applies) && return mix(SimpleMixing(α=mixing.α), basis, δF)
 
     # Solve J δρ = δF with J = (1 - χ0 vc) and χ_0 given as the sum of the χ0terms
-    devec(x) = reshape(x, size(δF_tot)..., n_spin)
-    function Jop(x)
-        δF = devec(x)  # [:, :, :, 1] is spin-up (or total), [:, :, :, 2] is spin-down
-
+    devec(x) = reshape(x, size(δF))
+    function Jop(δρ)
+        δρ = devec(δρ)
         # Apply Kernel (just vc for RPA and (vc + K_{xc}) if not RPA)
-        @views if n_spin == 1
-            x_tot  = from_real(basis, δF[:, :, :, 1])
-            x_spin = nothing
-        else
-            x_tot  = from_real(basis, δF[:, :, :, 1] .+ δF[:, :, :, 2])
-            x_spin = from_real(basis, δF[:, :, :, 1] .- δF[:, :, :, 2])
-        end
-        δV = apply_kernel(basis, x_tot, x_spin; ρ=ρin, ρspin=ρ_spin_in, RPA=mixing.RPA)
-
-        # set DC of δV to zero (δV[1] is spin-up or total, δV[2] is spin-down)
-        δV_DC = mean(mean(δV[σ].real) for σ in 1:n_spin)
-        δV[1].real .-= δV_DC
-        n_spin == 2 && (δV[2].real .-= δV_DC)
-
-        JδF = copy(δF)
+        δV = apply_kernel(basis, δρ; ρ=ρin, RPA=mixing.RPA)
+        δV .-= mean(δV)
+        Jδρ = copy(δρ)
         for apply_term! in χ0applies
-            apply_term!(JδF, δV, -1)  # JδF .-= χ0 * δV
+            apply_term!(Jδρ, δV, -1)  # JδF .-= χ0 * δV
         end
-        vec(JδF .-= mean(JδF))  # Zero DC component in total density response
+        Jδρ .-= mean(Jδρ)
+        vec(Jδρ)
```

Comment:
why not keep it as one line?

Reply:
why make it as one line? We literally never do this